### DISCUSSION TITLE
Boundary condition / by product of moving window

### DISCUSSION THREAD

**Discussion by cooluser26** [epochpic/epoch#611] [Q&A]
Hello everyone,

I'm using the EPOCH moving window in 1D. I'm injecting two monoenergetic beams of deuteron and triton into a uniform plasma (in the x direction). Throughout the simulation, I noticed that the B-field in the y&z direction starts growing from the right boundary of the simulation. Is this a genuine instability effect or a common by-product of the moving window?

The plot is attached below. Thank you!

![image](https://github.com/Warwick-Plasma/epoch/assets/108578700/bbc32a9a-1f26-4228-a51a-1c3c88e69854)

**Comment by DanRRRR** [comment]
As usually posting the settings file input.deck would help for others to try and see what might potentially happened. I saw this in incorrectly set window

**Reply by cooluser26** [reply]
Hi Dan,

The input deck I used is attached below. Is there an obvious hint as to why it led to the behavior seen? Thank you!

```
begin:control
  
   nx = 180*100*2

   x_min = 0
   x_max = 180e-6

   # Final time of simulation. 
   t_end = 1.8e-11
   smooth_currents = F
   stdout_frequency = 1000
end:control


begin:boundaries
   bc_x_min = periodic  
   bc_x_max = periodic 
end:boundaries


begin:constant
   dens_background = 1e+27
   yield = 8.4e+10
   r = 100e-6 
   dens_beam = yield/(4/3 * pi  *r^3)
   
   n_t = 80

   mass_ratio = 1836
   
   
   drift_deuteron = 9.8*1e-20
   drift_triton = 1.1*1e-19

   drift_electron = 0
   drift_silicon = 0
   drift_oxygen = 0
     
   temp_ev_beam = 0
   temp_ev_background = 2.4*1e+3

   MeV = qe* 1e+6
   m_d = 2*1.67e-27
   m_t = 3*1.67e-27
   pmax = sqrt(12.5*MeV*2*m_d)

   ppc = 500
end:constant


begin:window
  move_window = T
  #deuteron moves at 2.9e+7 and triton at 2.2e+7
  window_v_x = 2.0e+7
  window_start_time = 0
  bc_x_min_after_move = simple_outflow
  bc_x_max_after_move = simple_outflow
end:window


#######################
#### SPECIES BLOCK ####
#####################


begin:species
   # Rightwards traveling deuterons
   name = deuteron
   charge = +1
   mass = 2*mass_ratio
   temp_ev = temp_ev_beam
   drift_x = drift_deuteron
   number_density = if(x lt 5e-6, dens_beam, 0)
   npart_per_cell = ppc
   bc_x_min = simple_outflow
   bc_x_max = simple_outflow
end:species


begin:species
   # Rightwards traveling tritons
   name = triton
   charge = +1
   mass = 3*mass_ratio
   temp_ev = temp_ev_beam
   drift_x = drift_triton
   number_density = if(x lt 5e-6, dens_beam, 0)
   npart_per_cell = ppc
   bc_x_min = simple_outflow
   bc_x_max = simple_outflow
end:species


begin:species
   # background electrons
   name = background_electron
   charge = -1
   mass = 1
   temp_ev= temp_ev_background
   drift_x = drift_electron
   number_density = dens_background
   npart_per_cell = ppc
   bc_x_min = thermal
   bc_x_max = thermal
end:species


begin:species
   # background silicon/ablator ion. Assume v = 0 
   name = silicon
   charge = +14
   mass = 28*mass_ratio
   temp_ev= temp_ev_background
   drift_x = drift_silicon 
   number_density = dens_background*14 / (14+8)
   npart_per_cell = ppc
   bc_x_min = thermal
   bc_x_max = thermal
end:species

begin:species
   # background oxygen/ablator ion. Assume v = 0 
   name = oxygen
   charge = +8
   mass = 16*mass_ratio
   temp_ev= temp_ev_background
   drift_x = drift_oxygen
   number_density = dens_background*8 / (14+8)
   npart_per_cell = ppc
   bc_x_min = thermal
   bc_x_max = thermal
end:species

######################
#######PROBE BLOCK####
######################

begin:probe
   name = triton_0_probe
   point = 1e-18
   normal = (1)

   ek_min = 0.0
   ek_max = -1.0
   include_species : triton
   dumpmask = always
end:probe


begin:probe
   name = triton_350_probe
   point = 350e-6
   normal = (1)

   ek_min = 0.0
   ek_max = -1.0
   include_species : triton
   dumpmask = always
end:probe

begin:probe
   name = deuteron_0_probe
   point = 1e-18
   normal = (1)

   ek_min = 0.0
   ek_max = -1.0
   include_species : deuteron
   dumpmask = always
end:probe


begin:probe
   name = deuteron_350_probe
   point = 350e-6
   normal = (1)  

   ek_min = 0.0
   ek_max = -1.0
   include_species : deuteron
   dumpmask = always
end:probe

######################
### OUTPUT BLOCK  ####
######################
begin:output
   # Number of timesteps between output dumps
   dt_snapshot = t_end/n_t

   # Properties at particle positions
   particles = always
   
   vx= always          
   vy= always          
   vz= always 

   # Properties on grid
   grid = always
   ex= always
   ey = always
   ez = always

   bx= always
   by = always
   bz = always
   
   temperature = always + species
   particle_probes = always 
   charge_density = always + species
   number_density = always + species
     
end:output
```

**Comment by DanRRRR** [comment]
Sorry, i had no time to look at your case because still have not implemented the capability to output data in needed  me form (binary or ASCII not the SDF, and there is no SDF to regular ASCII or binary data converters. Would be nice if somebody in this community wrote this with Matlab or Python)  from 1D EPOCH.  But where i'd look first is your moving window parameters. Your particles have speed larger than the moving window and hence at some point will go into boundary conditions of EPOCH. My impression is that something  wrong happening then. I saw similar effect with this case #592. See the picture there, plasma parameters also get crazy at front boundary when the laser (while you have ion particles instead of laser) reach the boundary. You can see this effect already started to appear there. Next moments after that it will blow the entire simulation into pieces while by common sense the laser should just leave the plasma without such cataclysms  :).  Try first to increase the speed of moving window so that neither stream reach the boundary with time to avoid this effect

**Comment by Status-Mirror** [comment]
Hey @wnguyen1312,

I don't think the problem should be caused by `deuteron` or `triton` hitting `x_max`, because the macro-particles won't reach the moving `x_max` by 18 ps, even though they're moving faster than the moving window.

However, I'm also a little confused by the axes on your plot. If the `x_max` boundary starts at $180 \mu m$, and the window moves at $2\times 10^7 ms^{-1}$, then I'd expect `x_max` at 18 ps to be $540\mu m$ instead of the roughly $330 \mu m$ shown in your plot. If your plot is accurate, then maybe the isotopes _do_ exit the simulation?

There will be a field discontinuity in the newly loaded cells, as the fields here are set to the fields at the start of the simulation, which are usually 0. I haven't noticed this causing issues in other moving-window simulations, but I usually see people run laser-wakefield simulations with moving windows, which have lower plasma densities and massive laser fields present which might hide the problem. Your higher density plasma will have more numerical noise, leading to higher field noise, possibly seeding the numerical instability here.

If you want to convince yourself this is a moving window issue, try starting with a larger simulation window, and see if the instability starts at the same point in space.

Hope this helps,
Stuart

**Comment by DanRRRR** [comment]
Yea,  the density here indeed is very large to not to cause self-heating and instabilities on such time scales
